docs/Components.md

```diff
@@ -109,9 +109,25 @@ Assuming `navigator.language` is `"fr"`:
 <div><span>mardi 5 avril 2016</span></div>
 ```
 
-#### Multiple Intl Contexts
+### `RawIntlProvider`
```

Comment:
Naming is hard, but raw doesn’t seem to convey the meaning of this component. Is there a better name we could use? What if we just used IntlProvider and passed this indirectly?

Reply:
I'm open to suggestion, but I tried passing in `intl` to `IntlProvider` and it gets complicated bc we don't know which one to use (the pre-created `intl` or creating a new one). It also opens surface to unnecessary re-render since comparing `intl` object fails shallow equals easily.

Reply:
Is that possible to unify `IntlProvider` and `RawIntlProvider`. Won't a simple `SuperIntlProvider` as the super set does that?  I mean we can check if some method existed or not to decide which Provider to use internally, we don't need to expose this complexity to the user and blow up the API surface, IMHO.

Reply:
So the situation I was running into is, e.g if u specify:
```tsx
<IntlProvider intl={previouslyCreatedIntl} locale="fr" timeZone="America/New_York" />
```
what is supposed to happen? Since `previouslyCreatedIntl` was created with an embedded `locale` & basically a compiled object, we can't selectively override `locale` and other attributes. This creates a lot of footguns and TS right now doesn't allow you to have mutually exclusive attribute set (e.g if u pass in `intl`, don't pass in anything else). I agree that exposing the new API isn't great but this is for advanced use cases where you can pre-load Intl before rendering (in Electron/SSR) so not really targeting towards regular users. Hope that helps!